components/constellation/constellation.rs

```diff
@@ -1068,6 +1068,32 @@ where
                                 load_data,
                                 replace,
                             );
+                        } else {
+                            let pipeline_is_top_level_pipeline = self
+                                .browsing_contexts
+                                .get(&BrowsingContextId::from(top_level_browsing_context_id))
+                                .map(|ctx| ctx.pipeline_id == pipeline_id)
+                                .unwrap_or(false);
+                            // If the navigation is refused, and this concerns an iframe,
+                            // we need to take it out of it's "delaying-load-events-mode".
+                            // https://html.spec.whatwg.org/multipage/#delaying-load-events-mode
+                            if !pipeline_is_top_level_pipeline {
+                                let msg = ConstellationControlMsg::StopDelayingLoadEventsMode(
```

Comment:
@paulrouget I've changed this to sending a dedicated message, as opposed to re-use `handle_subframe_loaded`, since I think that would have resulted in double load events for the iframe...